var/spack/repos/builtin/packages/cube/package.py

```diff
@@ -46,16 +51,19 @@ class Cube(Package):
     variant('gui', default=False, description='Build CUBE GUI')
 
     depends_on('zlib')
-    depends_on('qt@4.6:', when='+gui')
 
-    def install(self, spec, prefix):
-        configure_args = ["--prefix=%s" % prefix,
-                          "--without-paraver"]
+    depends_on('qt@5:', when='@4.3 +gui')
+    depends_on('qt@4.6:', when='@4.2 +gui')
```

Comment:
This won't do what you think it should. When you say `@4.3`, spack will only apply this for _exactly_ 4.3. I think this is something we should consider fixing in the concretizer, but in the meantime, to get what you want, use `@4.3.0:4.3.999`.